src/java/ognl/OgnlRuntime.java

```diff
@@ -2076,11 +2076,26 @@ public static Object getStaticField(OgnlContext context, String className, Strin
                 }
             }
 
-            Field f = c.getField(fieldName);
+            Field f = getField(c, fieldName);
+            if (f == null) {
+                throw new NoSuchFieldException(fieldName);
+            }
             if (!Modifier.isStatic(f.getModifiers()))
                 throw new OgnlException("Field " + fieldName + " of class " + className + " is not static");
 
-            return f.get(null);
+            Object result = null;
+            if (!f.isAccessible() && context.getMemberAccess().isAccessible(context, null, f, null)) {
+                context.getMemberAccess().setup(context, null, f, null);
+                try {
+                    result = f.get(null);
+                } finally {
+                    context.getMemberAccess().restore(context, null, f, null, null);
```

Comment:
```suggestion
                    context.getMemberAccess().restore(context, null, f, null, state);
```

Restore the previous state